Uchu.Master/MasterServer.cs

```diff
@@ -82,6 +89,53 @@ private static async Task Main(string[] args)
             
             await SetupApiAsync();
             
+            // Setup health checks and automatic server closing
+            InstanceHeartBeatCheck = new Timer(WorldServerHeartBeatsIntervalInMinutes * 60000);
+            InstanceHeartBeatCheck.Elapsed += async (sender, eventArgs) =>
+            {
+                foreach (var instance in Instances.Where(i => i.ServerType == ServerType.World).ToList())
+                {
+                    var id = instance.Id.ToString();
+                    if (InstanceHealth.ContainsKey(id) && InstanceHeartBeats.ContainsKey(id))
+                    {
+                        InstanceHealth[id] = (InstanceHeartBeats[id] != 0 
+                                ? (float) InstanceHeartBeats[id] / WorldServerHeartBeatsPerInterval : 0) switch
+                        {
+                            var health when health >= 1 => ServerHealth.Healthy,
+                            var health when health >= 0.75 => ServerHealth.Lagging,
+                            var health when health >= 0.5 => ServerHealth.SeverelyLagging,
+                            var health when health >= 0.25 => ServerHealth.Unhealthy,
+                            
+                            // If hardly any heartbeats were received, gradually downgrade the health until closing
+                            _ => (ServerHealth) (int) InstanceHealth[id] - 1
+                        };
+                    }
+                    else
+                    {
+                        InstanceHealth.Add(id, ServerHealth.Dead);
```

Comment:
If an instance doesn't have a known health yet it's set to Dead here, but in the next `foreach` all Dead instances are killed. I might be missing something obvious here but why wouldn't all new instances be killed as soon as this check runs?

Reply:
This is just the case for a world that exists but for which the health is unknown, this does not actually create a new world, it's indeed intended to automatically kill a world for which the status is not known. When a new world is actually instanced (see MasterServer.cs, line 385), its status is set to healthy.

Reply:
That clears it up, thanks!